rest/main.go

```diff
@@ -369,28 +369,63 @@ func backupCurrentConfigFile(sourcePath string) (string, error) {
 }
 
 func CreateBootstrapConnectionFromStartupConfig(ctx context.Context, config *StartupConfig, bucketConnectionMode base.BucketConnectionMode) (base.BootstrapConnection, error) {
-	return createBootstrapConnectionWithCredentials(ctx, config, config.Bootstrap.Username, config.Bootstrap.Password, bucketConnectionMode)
+	opts := bootstrapConnectionOpts{
+		server:               config.Bootstrap.Server,
+		username:             config.Bootstrap.Username,
+		password:             config.Bootstrap.Password,
+		x509CertPath:         config.Bootstrap.X509CertPath,
+		x509KeyPath:          config.Bootstrap.X509KeyPath,
+		caCertPath:           config.Bootstrap.CACertPath,
+		bucketConnectionMode: bucketConnectionMode,
+		isServerless:         config.IsServerless(),
+	}
+	return createBootstrapConnectionWithOpts(ctx, opts)
 }
 
-func createBootstrapConnectionWithCredentials(ctx context.Context, config *StartupConfig, username, password string, bucketConnectionMode base.BucketConnectionMode) (base.BootstrapConnection, error) {
-	if base.ServerIsWalrus(config.Bootstrap.Server) {
-		cluster := base.NewRosmarCluster(config.Bootstrap.Server)
+type bootstrapConnectionOpts struct {
```

Comment:
Can we just have this use `CouchbaseClusterOptions`? Or document why these need to be different structs?

Reply:
Intermediate commit - I'd already realised I can get rid of `CouchbaseClusterOptions` which should resolve this